### ISSUE TITLE
Editing Openstack provider settings is ignoring application credentials

### ISSUE THREAD

**Issue by embik** [kubermatic/dashboard#4702]
### What happened

When using Openstack application credentials, the "Edit Provider" dialog from cluster details is not correct. It only provides options for username and password, which are unused fields when using Openstack application credentials. This is what it looks like currently:

![Screenshot 2022-07-21 at 09 12 27](https://user-images.githubusercontent.com/10295525/180152481-f82fc9fc-f091-4c6a-bb4a-3a4dae54d224.png)

When creating a cluster, application credentials are offered as option:

![Screenshot 2022-07-21 at 09 16 22](https://user-images.githubusercontent.com/10295525/180153180-86e14769-6429-4756-b552-079088f3c78c.png)

![Screenshot 2022-07-21 at 09 13 50](https://user-images.githubusercontent.com/10295525/180152752-5db044b2-2d9c-4ff9-abb0-f76bff390cdf.png)

After creating a cluster with application credentials, I tried to rototate/update credentials, for example because they are going to expire soon. There's several issues with that:

- I cannot update application credentials as only user credentials (username / password) are offered to be updated.
- It's possible on the backend to switch from app credentials to user credentials (and there's not really a reason to block that), but app credentials don't have a project. That means when switching from app credentials to user credentials, a third entry needs to be offered for the relevant project name or project ID (just like the setup dialog does).
- The current dialog, when used on an app credentials cluster, will render the cluster unusable, because no project will be set (also see https://github.com/kubermatic/kubermatic/issues/8809).

I think it's possible that we need some kind of API endpoint that returns what kind of credentials are used for the cluster (app creds or user creds) so the dashboard can understand what credentials were previously configured. Please open a kubermatic/kubermatic issue with what you need, as I'm not sure about that.

### Expected behavior

The dashboard knows what kind of credentials were used or at least offers the same options as the setup dialog, specifically resolving the bug that going from app credentials to user credentials via the current dialog will not ask for project information. Note that when updating existing user credentials, the project reference shouldn't be editable, since we don't support switching between projects and existing resources will be orphaned in the old project.

### How to reproduce

Create an Openstack cluster and try to rotate/update credentials.

### Environment
- UI Version: master
- API Version: master
- Domain:
- Others:

### Current workaround

Edit cluster credentials in the backend via `kubectl`. It also should work via the KKP API directly since the restriction on what fields can be updated is in the dashboard.

### Affected user persona

KKP users who own Openstack clusters

**Comment by Waseem826** [comment]
@embik do we need this fixed in 2.21?

**Comment by embik** [comment]
As far as I know this was reported by a user originally but @csengerszabo should probably decide on priority. IMHO it should be fixed since using the current dialog will break the cluster.

**Comment by Waseem826** [comment]
@embik since currently there's no way to determine credentials type in the UI so I've added a warning message when project is selected. Wdyt?

![screenshot-localhost_8000-2022 08 19-01_39_46](https://user-images.githubusercontent.com/13975988/185493029-e166ece7-aa5e-4ca1-85be-c900a1b0d6bc.png)

**Comment by embik** [comment]
@Waseem826 I think we should rephrase that because switching projects is really not supported/intended, but I can't think of anything better right now.

**Comment by Waseem826** [comment]
@embik so should I remove this warning message or keep it as it is?

**Comment by embik** [comment]
@Waseem826 can we rephrase it to "When switching from application to user credentials, ensure you are configuring the project already in use" maybe? Or something like it, if that is too long. But I think we should give clearer notes on when to consider this.

**Comment by Waseem826** [comment]
Updated

![screenshot-localhost_8000-2022 08 19-14_43_04](https://user-images.githubusercontent.com/13975988/185592159-11ca4743-2dcd-47c9-a435-0b1fbb737780.png)
